### ISSUE TITLE
Add new phase of Box production -- Research

### ISSUE THREAD

**Issue by maebeale** [rubyforgood/voices-of-consent#209]
Voices of Consent compiles resource materials specific to each person's BoxRequest data. This often requires referencing many websites, copy/pasting info into documents, adding some images/styling to make it prettier, and then saving the result as a PDF. We'd love to support/assist with that process later on, and this is the initial groundwork to add Research as a phase of Box production. Technically it's required for specific BoxItems (we don't need research on a "toothbrush", but we do on a "local resources packet"). The Researcher is often a different person than the Assembler. Once the packet PDF has been uploaded to the BoxItem, the Assembler will print the item before marking it off on the checklist of included items as part of BoxAssembly (packaging).

- After a Box is Designed and before it is Assembled/Packaged, there is a Research phase
- This will need to be added to Box/BoxItem aasm
- New controller and view will be needed as React components
- Also, a field should be added to BoxItem: `requires_research` (in order to trigger that Box as needing research, bc one of the BoxItems added had a research component)
- Once that BoxItem has a file uploaded, we know the research has been completed
- Later we will add ways to help the person do the research

**Comment by maebeale** [comment]
@MxOliver is there any chance you might be interested in this issue bc of its ties in to aasm? the above is a suggested approach -- happy to talk more about best implementation if you were interested in taking this on and/or splitting it up into back end and front end issues.

**Comment by MxOliver** [comment]
Yes I can absolutely work on this, I'll ping you on slack when I get started to talk about implementation details

**Comment by maebeale** [comment]
Yay! Thank you!

**Comment by maebeale** [comment]
@MxOliver  I'm working through lifecycle implementation for the current actions in the controllers and UI, but it hasn't impacted the current aasm stuff much. Just wanted to send a head's up.

**Comment by maebeale** [comment]
@MxOliver I've added a bunch of this bc I was in there anyway trying to get all the workflow stuff working with the views & controllers

**Comment by MxOliver** [comment]
Awesome! sorry that I'm a bit slow on these things I generally only have the weekend

**Comment by maebeale** [comment]
Oh! No worries AT ALL! I just got deep in there and figured it was worth it to just take a rough first pass while getting everything working the same across the life cycle. I was worried it would feel like a referendum, so REALLY hope that wasn't the case! There are tons of small cleanups and tests to do for all the ways I expanded on your aasm work, and I'd LOVE your review of what I did. There's 0% chance I could've done it without your initial work on the state machine, so, thank you so much!
